larray/core/metadata.py

```diff
@@ -8,40 +8,8 @@
 __all__ = ['Metadata']
 
 
-class BaseMetadata(OrderedDict):
-    """
-    Base class for AttributeDict subclass.
-
-    Examples
-    --------
-    >>> import os.path
-    >>> from datetime import datetime
-    >>> from larray.core.array import ndtest
-    >>> tmpdir = getfixture('tmpdir')
-    >>> attrs = Metadata([('string', 'a string'), ('number', 5), ('float', 5.6), ('time', datetime(2017, 2, 10))])
-
-    Save/Load using HDF format
-
-    >>> from larray.util.misc import LHDFStore
-    >>> fname = os.path.join(tmpdir.strpath, 'metadata.h5')
-    >>> key = 'arr'
-    >>> with LHDFStore(fname) as store:
-    ...     # save
-    ...     ndtest(3).to_hdf(store, key)
-    ...     attrs.to_hdf(store, key)
-    ...     # load
-    ...     attrs2 = BaseMetadata.from_hdf(store, key)
-    >>> attrs2
-    string: a string
-    number: 5
-    float: 5.6
-    time: 2017-02-10 00:00:00
-    """
-    def __init__(self, *args, **kwargs):
-        super(BaseMetadata, self).__init__(*args, **kwargs)
-
-    def __repr__(self):
-        return '\n'.join(['{}: {}'.format(k, v) for k, v in self.items()])
+class BaseMetadata(object):
```

Comment:
Wouldn't it be cleaner to define an AttributeDict class (differently on PY2 and PY3) then have a single Metadata class (for both PY2 and PY3) inheriting from it?

Reply:
IO functions will be the same for PY2 and PY3

Reply:
yes, and?

Reply:
we will have duplicated code

Reply:
no, since the io code would only be in the Metadata class.